src/server/aws_lambda.ts

```diff
@@ -18,46 +19,48 @@ function isHandlerSync(handler: Handler): handler is SyncHandler {
     return handler.length > 2
 }
 
-function asyncHandler<TEvent = any, TResult = any>(handler: AsyncHandler<TEvent, TResult>, hb: Honeybadger): AsyncHandler<TEvent, TResult> {
-    return async (event, context) => {
-        try {
-            return await handler(event, context)
+function reportToHoneybadger(hb: Honeybadger, err: Error | string | null, callback: (err: Error | string | null) => void) {
+    hb.notify(err, {
+        afterNotify: function () {
+            hb.clear()
+            callback(err)
         }
-        catch (err) {
-            return new Promise((_, reject) => {
-                hb.notify(err, {
-                    afterNotify: function () {
-                        hb.clear()
-                        reject(err)
-                    }
-                })
+    })
+}
+
+function asyncHandler<TEvent = any, TResult = any>(handler: AsyncHandler<TEvent, TResult>, hb: Honeybadger): AsyncHandler<TEvent, TResult> {
+    return (event, context) => {
+        hb.configure({store: AsyncStore})
+        return new Promise<TResult>((resolve, reject) => {
+            AsyncStore.run({context: {}, breadcrumbs: []}, () => {
```

Comment:
I know that this could be [awaited](https://nodejs.org/api/async_context.html#usage-with-asyncawait) but I didn't find a way to make it work with Typescript (return type is `void`).

Reply:
Should this not be `store.run()`? Looks like `AsyncStore` is the type, not the object.

Reply:
Ah, my bad; just realised it is the object. Perhaps we should rename the exported constant to `asyncStore`, so it's clear it's an object, not a type or class.

Reply:
Poking @joshuap on this also.

Reply:
@subzero10 I think I've been capitalizing types, classes, interfaces, etc. ([example](https://github.com/honeybadger-io/honeybadger-js/blob/master/src/browser.ts#L1)), and lower casing functions ([example](https://github.com/honeybadger-io/honeybadger-js/blob/d792a67a465089f31d4f62f4b83190dacfe56580/src/browser.ts#L4)). I don't have a strong opinion on the convention as long as we're consistent.

Reply:
OK, let me clarify this a little more:
- Class name: PascalCase
- function name: camelCase
- Singleton Instances of Classes: This is the case where a class is instantiated once and used throughout the app, like the Honeybadger instance [here](https://github.com/honeybadger-io/honeybadger-js/blob/master/examples/aws-lambda/honeybadger.js#L1) and [here](https://github.com/honeybadger-io/honeybadger-js/blob/master/src/server.ts#L108). I followed this pattern and did the same for `AsyncStore`.

Reply:
OK, thanks for clarifying. For `AsyncLocalStorage`, I went with a singleton because that's how all the examples I went through do it. Plus, I also tried creating a new ALS instance on for each lambda handler invocation and the tests were failing. That was at the beginning of the implementation though and I didn't dive deep into it.

Reply:
**Update**
I just tried without a singleton:
```
const store = new AsyncStore()
hb.configure({store})
return new Promise<TResult>((resolve, reject) => {
    store.run({context: {}, breadcrumbs: []}, () => {
        ...
    })
})
```
This doesn't work because `hb.configure({ store })` every time sets the `store` in `Honeybadger` to a new ALS object. When running the lambda handlers at the same time, only the handler that set the `store` last will have a context. The other handler(s) returns undefined and fails the test.
Note: I know that lambda handlers running simultaneously is unlikely but I think its better (and future-proof) to cover this scenario.

When `AsyncLocalStorage` is a singleton, `hb.configure({ store })` always sets to the same object and all handlers retrieve the correct context (because they are inside the callback of the `run` method).

So, based on our implementation, I think there is good reason to keep a singleton for ALS.